binding/python/_orca.py

```diff
@@ -0,0 +1,396 @@
+    def max_character_limit(self) -> int:
+        """Maximum number of characters allowed in a single synthesis request."""
+
+        return self._max_character_limit
+
+    def synthesize(
+            self,
+            text: str,
+            speech_rate: Optional[float] = None) -> Sequence[int]:
+        """
+        Generates audio from text. The returned audio contains the speech representation of the text.
+
+        :param text: Text to be converted to audio. The maximum number of characters per call to `.synthesize()` is
+        `.max_character_limit()`. Allowed characters are lower-case and upper-case letters and punctuation marks
+        that can be retrieved with `.valid_punctuation_symbols()`.
+        Custom pronunciations can be embedded in the text via the syntax `{word|pronunciation}`.
+        The pronunciation is expressed in ARPAbet format, e.g.: "I {live|L IH V} in {Sevilla|S EH V IY Y AH}".
+        :param speech_rate: Rate of speech of the synthesized audio.
+        :return: The generated audio, stored as a sequence of 16-bit linearly-encoded integers.
+        """
+
+        c_synthesize_params = self._get_c_synthesize_params(speech_rate=speech_rate)
+
+        c_num_samples = c_int32()
+        c_pcm = POINTER(c_int16)()
+        status = self._synthesize_func(
+            self._handle,
+            text.encode("utf-8"),
+            c_synthesize_params,
+            byref(c_num_samples),
+            byref(c_pcm))
+        if status is not self.PicovoiceStatuses.SUCCESS:
+            raise self._PICOVOICE_STATUS_TO_EXCEPTION[status](
+                message="Unable to synthesize speech",
+                message_stack=self._get_error_stack())
+
+        pcm = [c_pcm[i] for i in range(c_num_samples.value)]
+
+        self._delete_pcm_func(c_pcm)
+        self._synthesize_params_delete_func(c_synthesize_params)
+
+        return pcm
+
+    def synthesize_to_file(
+            self,
+            text: str,
+            output_path: str,
+            speech_rate: Optional[float] = None) -> bool:
+        """
+        Generates audio from text. The returned audio contains the speech representation of the text.
+
+        :param text: Text to be converted to audio. The maximum number of characters per call to `.synthesize()` is
+        `.max_character_limit()`. Allowed characters are lower-case and upper-case letters and punctuation marks
+        that can be retrieved with `.valid_punctuation_symbols()`.
+        Custom pronunciations can be embedded in the text via the syntax `{word|pronunciation}`.
+        The pronunciation is expressed in ARPAbet format, e.g.: "I {live|L IH V} in {Sevilla|S EH V IY Y AH}".
+        :param output_path: Absolute path to the output audio file. The output file is saved as `WAV (.wav)`
+        and consists of a single mono channel.
+        :param speech_rate: Rate of speech of the generated audio.
+        :return: True if the operation is successful, throws an exception otherwise.
```

Comment:
if it's either true or exception, it sounds like this function should just be void because you essentially know that it succeeded if it returned successfully

Reply:
Right, made it void!